tokio/src/net/windows/named_pipe.rs

```diff
@@ -2626,7 +2626,7 @@ pub enum PipeEnd {
 /// Information about a named pipe.
 ///
 /// Constructed through [`NamedPipeServer::info`] or [`NamedPipeClient::info`].
-#[derive(Debug)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
 #[non_exhaustive]
 pub struct PipeInfo {
```

Comment:
This type is `#[non_exhaustive]` so that we can add additional fields in the future. Adding `Copy` goes against that since it prevents the addition of non-copy fields in the future.

The same could apply to the other traits, but in a lesser extent.

Reply:
Since this is only constructed from calling NamedPipe[Server|Client] and those are convenience methods to call [GetNamedPipeInfo](https://learn.microsoft.com/en-us/windows/win32/api/namedpipeapi/nf-namedpipeapi-getnamedpipeinfo), I wouldn't expect tokio to ever return a different substantially different struct than the one from the win32 api. With that in mind, I don't think `#[non_exhaustive]` is super useful here.

But if that's a dealbreaker, would this change be acceptable without `Copy`? Or at least with only the addition of `Clone`?

Reply:
OS apis such as `GetNamedPipeInfo` are sometimes extended to return additional data in newer versions of the OS.

The change is acceptable with just Clone.